### PR TITLE
tags endpoint and support querying by tag via SQL interface

### PR DESCRIPTION
**Pull request by tap2k**
- Support new Tags endpoint for retrieving tags (optionally by prefix for autocomplete)
- Support querying objects by Tag using the SQL interface
- Support exploratory new endpoint for retrieving all objects according to Tag
- Make single quotes optional in SQL interface

### CODE REVIEW

File: apps/site/api/urls.py [hunk 1]
```diff
@@ -135,9 +135,18 @@
         r'^photos/(?P<pk>[0-9]+)/rotate-right/$',
         views.rotate_right,
         name='rotate-right'),
+    url(r'^tags/(?P<term>\w+)/$',
+        views.TagList.as_view(),
```
Comment (vanwars) [inline apps/site/api/urls.py]: This new endpoint needs tests.

File: apps/site/api/urls.py [hunk 2]
```diff
@@ -135,9 +135,18 @@
...
+        name='tag-list'),
     url(r'^tags/$',
         views.TagList.as_view(),
         name='tag-list'),
+    url(r'^searchbytag/(?P<tag>\w+)/$',
+        views.TagSearchList.as_view(),
+        name='tag-search-list'),
+    url(r'^searchbytag/$',
+        views.TagSearchList.as_view(),
+        name='tag-search-list'),
```
Comment (vanwars) [inline apps/site/api/urls.py]: searchbytag endpoints needs tests

**Comment by vanwars** [comment]
Another question re: single quotes. I think we should we be supporting both single quotes and no single quotes, for example:
* where name = 'purple cat'
* where name like '%purple ca%'
* where name = purple cat
* where name like %purple ca%

It appears as if the code does support both permutations, but can you enhance the tests in sql_parse_tests.py so that there is test coverage for both?

File: apps/site/api/urls.py (hunk 2 above)
Comment (tap2k) [inline apps/site/api/urls.py]: Im not sure if this is the API interface we want so I backed out this endpoint. Currently you can search by tag for specific media using the sql endpoint.

File: apps/site/api/urls.py (hunk 1 above)
Comment (tap2k) [inline apps/site/api/urls.py]: Done

**Comment by tap2k** [comment]
Done

**Comment by vanwars** [comment]
Hi Tap, almost there, but when adding [additional tests](https://github.com/LocalGround/localground/pull/65/commits/4700c87b0c8f5b74d6b1f2a3aea9a741426f5f3f), I managed to find a case where omitting the single quotes requirement produced an error. The error happens for text strings that start with a number, like this one:

```WHERE file_name_orig=2013-07-04 16.56.55.jpg```

which gets parsed as:

```(AND: ('file_name_orig__exact', '2013'))```

You can see for yourself by running:
```
python manage.py test localground.apps.site.api.tests.sql_parse_tests.SQLParseTest.test_equality_operator
```
Basically, in this file: localground/apps/lib/helpers/sqlparse.py,  the comparison_pattern method (around line 150) is trying to parse strings that start with numbers as numbers instead of strings. Can you fix the regex? Otherwise, looks good!

**Comment by tap2k** [comment]
OK should be fixed now!

**Pull request merged**